# openssl_extconf_ruby_export_msys_6.patch
#
# Target:  ext/openssl/extconf.rb (pass1 (msys))
# Source:  tebako gem lib/tebako/packager/pass1_patch.rb Pass1MSysPatch::OPENSSL_EXTCONF_RB_PATCH
#
# msys: RUBY_EXPORT=1 shall be set for static builds but is missing in the
# openssl extension; push -DRUBY_EXPORT=1 into $defs.
diff --git a/ext/openssl/extconf.rb b/ext/openssl/extconf.rb
index 06ed4f6..ac39d80 100644
--- a/ext/openssl/extconf.rb
+++ b/ext/openssl/extconf.rb
@@ -172,8 +172,13 @@ have_func("SSL_get0_peer_signature_name(NULL, NULL)", ssl_h)
 # added in 4.0.0
 have_func("ASN1_BIT_STRING_set1(NULL, NULL, 0, 0)", "openssl/asn1.h")
 
+# Start of tebako patch
+$defs.push("-DRUBY_EXPORT=1")
+# End of tebako patch
+
 Logging::message "=== Checking done. ===\n"
 
+
 # Append flags from environment variables.
 extcflags = ENV["RUBY_OPENSSL_EXTCFLAGS"]
 append_cflags(extcflags.split) if extcflags
